pkg/trigram/trigram_test.go

```diff
@@ -0,0 +1,159 @@
+package trigram
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestTrigram(t *testing.T) {
+	t.Run("test basic trigram", func(t *testing.T) {
+		p := "howdydo"
+
+		expected := [5]Trigram{
+			{
+				Word:   "how",
+				Offset: 0,
```

Comment:
It seems like it might be more useful to store the byte range instead of the byte offset, is the byte offset sufficient to create a `DataPointer`?

Reply:
This is the current scheme for inserting an element:

```go
for _, tri := range trigrams {
     valueBytes := []byte(tri.Word)
     mp.Offset += tri.Offset
     mp.Length += uint32(tri.Offset)
     if err := page.BPTree(&btree.BPTree{Data: r, DataParser: j, Width: width}).Insert(
          btree.ReferencedValue{
               DataPointer: mp,
               Value:       valueBytes,
          }, data); err != nil {
          return fmt.Errorf("failed to insert into b+tree: %w", err)
     }
}
```

Since the existing `mp.Offset` is ` data.Offset + uint64(fieldOffset)`, it makes sense to just add the trigram offset to the existing offset.

I do that with length as well, but I can see why we'd want to store the `Length` as well.

Since given `ho¥wdy` will create the following trigrams:

```typescript
[{ word: "how", offset: 0, length: 4 (this is because `ho¥w`}, ...]
```